controlplane/src/core/repositories/FeatureFlagRepository.ts

```diff
@@ -0,0 +1,1073 @@
+      if (!baseSubgraph) {
+        continue;
+      }
+
+      featureSubgraphs.push({
+        ...fs,
+        baseSubgraphId: f.baseSubgraphId,
+        baseSubgraphName: baseSubgraph.name,
+      });
+    }
+
+    return featureSubgraphs;
+  }
+
+  public async getFeatureSubgraphsCount({ namespaceId, query }: FeatureFlagListFilterOptions) {
+    const conditions: SQL<unknown>[] = [
+      eq(targets.organizationId, this.organizationId),
+      eq(targets.type, 'subgraph'),
+      eq(subgraphs.isFeatureSubgraph, true),
+    ];
+
+    if (namespaceId) {
+      conditions.push(eq(targets.namespaceId, namespaceId));
+    }
+
+    if (query) {
+      conditions.push(like(targets.name, `%${query}%`));
+    }
+
+    const featureSubgraphTargets = await this.db
+      .select({
+        count: count(),
+      })
+      .from(targets)
+      .innerJoin(subgraphs, eq(schema.subgraphs.targetId, schema.targets.id))
+      .where(and(...conditions));
+
+    return featureSubgraphTargets[0].count;
+  }
+
+  public async getFeatureFlagById({
+    featureFlagId,
+    namespaceId,
+  }: {
+    featureFlagId: string;
+    namespaceId: string;
+  }): Promise<FeatureFlagDTO | undefined> {
+    const resp = await this.db
+      .select({
+        id: featureFlags.id,
+        name: featureFlags.name,
+        namespaceId: featureFlags.namespaceId,
+        namespace: namespaces.name,
+        labels: featureFlags.labels,
+        creatorUserId: featureFlags.createdBy,
+        isEnabled: featureFlags.isEnabled,
+        organizationId: featureFlags.organizationId,
+        createdAt: featureFlags.createdAt,
+        updatedAt: featureFlags.updatedAt,
+      })
+      .from(featureFlags)
+      .innerJoin(namespaces, eq(namespaces.id, featureFlags.namespaceId))
+      .where(
+        and(
+          eq(featureFlags.organizationId, this.organizationId),
+          eq(featureFlags.id, featureFlagId),
+          eq(featureFlags.namespaceId, namespaceId),
+        ),
+      )
+      .execute();
+    if (resp.length === 0) {
+      return;
+    }
+
+    let createdBy = '';
+    if (resp[0].creatorUserId) {
+      const userRepo = new UserRepository(this.db);
+      const user = await userRepo.byId(resp[0].creatorUserId);
+      createdBy = user?.email || '';
+    }
+
+    const featureSubgraphs = await this.getFeatureSubgraphsByFeatureFlag({
+      namespaceId,
+      featureFlagId,
+    });
+
+    return {
+      ...resp[0],
+      labels: resp[0].labels?.map?.((l) => splitLabel(l)) ?? [],
+      createdAt: resp[0].createdAt.toISOString(),
+      updatedAt: resp[0].updatedAt?.toISOString() || '',
+      createdBy,
+      creatorUserId: resp[0].creatorUserId || undefined,
+      featureSubgraphs,
+    };
+  }
+
+  public async getFeatureFlagByName({
+    featureFlagName,
+    namespaceId,
+  }: {
+    featureFlagName: string;
+    namespaceId: string;
+  }): Promise<FeatureFlagDTO | undefined> {
+    const resp = await this.db
+      .select({
```

Comment:
Most of  this can be moved into a private function like how we do with FederatedGraph

Reply:
Done